### PR TITLE
Remove normalizeFilename function

### PR DESCRIPTION
**Pull request by ficristo**
Quick hack for https://github.com/adobe/brackets/issues/12744, I doubt this is the correct way to fix.
I am only interested to see if this could solve the crashes people are experiencing.

/cc @zaggino

### CODE REVIEW

**Comment by zaggino** [comment]
This construct
```
var fileName = fileInfo.name;
if (ternServer.projectDir && fileName.indexOf(ternServer.projectDir) === -1) {
  fileName = ternServer.projectDir + fileName;
}
```
repeats itself if I have counted right and not missed any difference 5 times over the diff, perhaps
```
function getNormalizedFilename(fileInfo, ternServer) {
  return ...
}
```
and use `var fileName = getNormalizedFilename(fileInfo, ternServer);` would be a bit nicer

**Comment by ficristo** [comment]
Sure! If this fixes the crashes I'll refactor like that.

**Comment by zaggino** [comment]
Anyway, were you able to replicate anything, or is this just a hinch? If it doesn't break any tests I think we could merge this for time being as https://github.com/adobe/brackets/pull/11948 doesn't seem to be moving much...

**Comment by ficristo** [comment]
Since people repored crashes in 1.8 that weren't in 1.8 this is the only thing that comes to my mind.
So yes, this is only an hinch...
The tests pass.

**Comment by zaggino** [comment]
I have a project which, while working on it, definitely crashes Brackets, but I can't replicate this. Sometimes after 5 minutes, sometimes after an hour. So it'd vote the approach of merging it and seeing if there's any change in the reports.

**Review by zaggino — CHANGES_REQUESTED** [review]
refactor before merge

**Comment by ficristo** [comment]
Refactored. After running tests again, they are quite flaky. But I'm not sure if it is because our testsuite or because my PR...

File: src/extensions/default/JavaScriptCodeHints/tern-worker.js
Comment (zaggino) [inline src/extensions/default/JavaScriptCodeHints/tern-worker.js]: if you don't mind one last change, for future readability, I'd extract this to `_getDenormalizedFilename` as this basically reverts `_getNormalizedFilename` if I'm not mistaken

**Comment by zaggino** [comment]
Hi @ficristo , I've run the tests and they all pass on my machine so that thing is fine.
Just one minor thing (commented above) and I'd merge this if nobody doesn't want to add anything.

File: src/extensions/default/JavaScriptCodeHints/tern-worker.js
Comment (ficristo) [inline src/extensions/default/JavaScriptCodeHints/tern-worker.js]: Done

**Comment by zaggino** [comment]
👍

**Pull request merged**
